pkg/service/aws/volume.go

```diff
@@ -317,3 +332,65 @@ func (a *awsController) DeleteSnapshot(ctx context.Context, req *proto.DeleteSna
 
 	return &proto.DeleteSnapshotResponse{}, nil
 }
+
+func (a *awsController) CopySnapshot(ctx context.Context, req *proto.CopySnapshotRequest) (*proto.CopySnapshotResponse, error) {
+
+	region := req.Region
+	snapshotId := req.SnapshotId
+	session, err := NewSession(ctx, region, req.AccountName)
+
+	if err != nil {
+		a.logger.Error(ctx, "Can't start AWS session", "error", err)
+		return nil, errors.Wrap(err, "CopySnapshot: failed to get session")
+	}
+
+	ec2Client := session.getEC2Client()
+
+	//check if snapshot exist
+
+	snap, err := ec2Client.DescribeSnapshotsWithContext(ctx, &ec2.DescribeSnapshotsInput{
+		SnapshotIds: []*string{&snapshotId},
+	})
+	if err != nil {
+		a.logger.Error(ctx, "failed to fetch the snapshot", "error", err, "snapshotid", snapshotId)
+		return nil, errors.Wrap(err, "faild to fetch source snapshot")
+	}
+
+	if len(snap.Snapshots) == 0 {
+		a.logger.Error(ctx, "failed to fetch the snapshot, got 0 snapshots in response", "snapshotid", snapshotId)
+		return nil, errors.New("CopySnapshot: failed to fetch the source snapshot")
+	}
+	copyDesc := fmt.Sprintf("copy of snapshot: %s", snapshotId)
+
+	name := fmt.Sprintf("copy-%s", snapshotId)
```

Comment:
Should we have this? we should probably not hardcode it like this, probably a tag would be better, for source snapshot id(and uri?) 
same comment for other providers